src/network.py

```diff
@@ -0,0 +1,33 @@
+'''
+Contains the code for the encoder & decoders, which are composed of modules.  
+These modules are pre, encoder, a decoder, a post, and optional discriminator.
+    - BiRNN encoder/decoder for text and speech (with attention (?))
+    - Transformer encoder/decoder for text and speech
+'''
+
+import torch.nn as nn
+
+class Discriminator(nn.Module):
+    # TODO: Fill in with linear layers :) 
+    def __init__(self):
+        super(Discriminator, self).__init__()
+        
+class SpeechTransformer(nn.Module):
+    # TODO: Fill in with pre/post needed and enc/dec
+    def __init_(self):
+        super(SpeechTransformer, self).__init__()
+    
+class TextTransformer(nn.Module):
+    # TODO: Fill in with pre/post needed and enc/dec
+    def __init_(self):
+        super(TextTransformer, self).__init__()
+
+class TextRNN(nn.Module):
+    # TODO: Fill in with pre/post needed and enc/dec
+    def __init_(self):
+        super(TextRNN, self).__init__()
+
+class SpeechRNN(nn.Module):
+    # TODO: Fill in with pre/post needed and enc/dec
+    def __init_(self):
+        super(SpeechRNN, self).__init__()
```

Comment:
Should not be separate speech/text modules. 
A complete model should have speech encoder, speech decoder, text encoder, text decoder, pre/post modules all from modules.py
, and discriminator from network.py.

Then the transformer model and rnn model should extend this base class.

Reply:
The reason I have the different speech/text modules here is because from my understanding they need different post/pre nets. I was thinking two parallel models (one speech and one text) would be easier to handle the having one complete model with both, but we can add another wrapper class that contains a Speech & Text model pair!

I like the idea of having the base class, I am on it

Reply:
I went ahead and simplified this a bit more

Reply:
That is true, one giant model is a bit confusing; however, I don't see how realistically these text/speech classes can be used for anything other than the denoising autoencoding task. We have to pass from text encoder to speech encoder or vice versa. Having this intermediary full model add another layer of abstraction, that we actually have to break apart to do any ASR or TTS.

Reply:
New changes look good. I think maybe having less components is easier to keep track of however, but as long as you can understand it, write the training code, and make it easy to switch between the RNN and Transformer models, it should be fine.